misc/multiaddr/src/protocol.rs

```diff
@@ -75,6 +76,7 @@ pub enum Protocol<'a> {
     /// Contains the "port" to contact. Similar to TCP or UDP, 0 means "assign me a port".
     Memory(u64),
     Onion(Cow<'a, [u8; 10]>, u16),
+    Onion3(Cow<'a, [u8]>, u16),
```

Comment:
I wonder if we should use `Onion3(Cow<'a, [u8; 35]>, u16)` instead? It would more than double the size of `Protocol` and require custom impls of `PartialEq` and `Debug` but the type would be more descriptive and we would not require a heap allocation when parsing `Protocol::Onion3`. What do you think?

Reply:
Hey, sorry was AFK - Agree with saving an allocation, I'll take another shot because I think I was rushed and just went with the easier option :)

Reply:
I'll make a custom type `Onion3Addr(([u8; 35], u16))` and implement `PartialEq` and `Debug`

Reply:
Done - let me know what you think :)